llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -679,6 +685,7 @@ void HWAddressSanitizer::initializeModule() {
 
   InstrumentGlobals =
       !CompileKernel && !UsePageAliases && optOr(ClGlobals, NewRuntime);
+  InstrumentAllGlobals = InstrumentGlobals && ClAllGlobals;
```

Comment:
The call to `instrumentGlobals` is already guarded by `InstrumentGlobals`. This is not needed.